server/docker/Dockerfile

```diff
@@ -1,5 +1,84 @@
-# Use Eclipse Temurin with Java 21 as the base image
-FROM eclipse-temurin:21
+########################################################################################################################
+#
+# Define Global Build Arguments
+#
+########################################################################################################################
+ARG UBUNTU_TAG="focal-20230605"
+
+########################################################################################################################
+#
+# Setup Builder Image
+#
+########################################################################################################################
+FROM ubuntu:${UBUNTU_TAG} AS openjdk-builder
+
+# Define Standard Environment Variables
+ENV LC_ALL=C.UTF-8
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install basic OS utilities for building
+RUN apt-get update && \
+	apt-get install --yes tar gzip gnupg2 curl
+
+##########################
+####    Java Setup    ####
+##########################
+
+RUN set -eux; \
+        ARCH="$(dpkg --print-architecture)"; \
+        case "${ARCH}" in \
+           aarch64|arm64) \
+             ESUM='e184dc29a6712c1f78754ab36fb48866583665fa345324f1a79e569c064f95e9'; \
+             BINARY_URL='https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1%2B12/OpenJDK21U-jdk_aarch64_linux_hotspot_21.0.1_12.tar.gz'; \
+             ;; \
+           amd64|i386:x86-64) \
+             ESUM='1a6fa8abda4c5caed915cfbeeb176e7fbd12eb6b222f26e290ee45808b529aa1'; \
+             BINARY_URL='https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1%2B12/OpenJDK21U-jdk_x64_linux_hotspot_21.0.1_12.tar.gz'; \
+             ;; \
+           ppc64el|powerpc:common64) \
+             ESUM='9574828ef3d735a25404ced82e09bf20e1614f7d6403956002de9cfbfcb8638f'; \
+             BINARY_URL='https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1%2B12/OpenJDK21U-jdk_ppc64le_linux_hotspot_21.0.1_12.tar.gz'; \
+             ;; \
+           *) \
+             echo "Unsupported arch: ${ARCH}"; \
+             exit 1; \
+             ;; \
+        esac; \
```

Comment:
These versions are well behind; and we _should not_ be using them.  We probably need to update the checksum and URL for the latest 21.0.5 build.
Also; is there a reason we don't use the `apt` or `dpkg` installs?

Reply:
Copy/paste from Nathan's code